test/test_type_promotion.py

```diff
@@ -218,6 +218,47 @@ def test_create_bool_tensors(self):
         # this seems like odd behavior but ints also create float tensors, numpy doesn't have this function.
         self.assertEqual(torch.scalar_tensor(False, device=self.device), torch.tensor(0., device=self.device))
 
+    def test_comparison_ops_with_type_promotion(self):
+        value_for_type = {
+            torch.uint8: (1 << 5),
+            torch.int8: (1 << 5),
+            torch.int16: (1 << 10),
+            torch.int32: (1 << 20),
+            torch.int64: (1 << 30),
+            torch.float16: (1 << 10),
+            torch.float32: (1 << 20),
+            torch.float64: (1 << 30)
+        }
+        comparison_ops = [
+            dict(
```

Comment:
nit: this is much nicer as a namedtuple

Reply:
Will fix it, when adding other comparison ops